app/controllers/rounds_controller.rb

```diff
@@ -39,6 +39,16 @@ def pick_agenda_cards
     redirect_to @current_game
   end
 
+  def reveal_objective
+    round = find_round
+    params[:game][:public_objectives].each do |key|
+      round.revealed_objectives << PublicObjective.find_by_key(key)
```

Comment:
Sorry, I didn't commented on this before :( But this must be changed as well to allow secret objectives :) Maybe it would be useful to move that logic from a view helper to the data layer 🤔 What do you think?